torchvision/prototype/transforms/_transform.py

```diff
@@ -37,9 +38,30 @@ def forward(self, *inputs: Any) -> Any:
 
         params = self._get_params(flat_inputs)
 
-        flat_outputs = [
-            self._transform(inpt, params) if check_type(inpt, self._transformed_types) else inpt for inpt in flat_inputs
-        ]
+        # This is a heuristic on how to deal with simple tensor inputs:
+        # 1. If we haven't seen an image yet, we transform a simple tensor
+        # 2. If we have seen an image before, so either a simple tensor, a `datapoints.Image` or a `PIL.Image.Image`,
+        #    we return simple tensors without modification.
+        # The order is defined by the returned list of `tree_flatten`, which recurses depth-first through the input.
+        # Since in most cases the image is the first input or at least comes before any other numerical data, this
+        # heuristic allows users to keep any supplemental numerical data in the sample as simple tensors. We have a few
+        # datasets, like `Caltech101`, `CelebA`, and `Widerface`, that would need special handling without this
+        # heuristic, since they return the target partially or completely as tensors.
+        # TODO: try to get user feedback if this heuristic is confusing or it is fine to keep it
```

Comment:
We have to be careful about how we frame the feedback request, if we do one. If we just ask "what do you think of this behaviour" the first reaction is likely going to be "this sounds weird af" and they'd be right.
It's important to note that 99% of users don't even need to care about that. The only users who need to know about this is those who are passing multiple tensors to the transforms and expect them to be treated as images, which is not even something we claim to support

Reply:
This was more a comment for us to not forget about this. But indeed, people might look at the code and we should phrase it better.